euphony/src/main/cpp/core/source/Base32.cpp

```diff
@@ -0,0 +1,80 @@
+#include "../Base32.h"
+#include <iomanip>
+#include <sstream>
+#include <cmath>
+
+using namespace Euphony;
+
+Base32::Base32(const HexVector &hexVectorSrc)
+: hexVector(hexVectorSrc) {}
+
+std::string Base32::getBaseString() {
+    std::stringstream ss;
+    int sum = 0;
+    const int bitDivisionUnit = 5;
+    int rest = hexVector.getSize() % bitDivisionUnit;
+    if(!rest) rest = bitDivisionUnit;
+
+    int count = 0;
+    for(u_int8_t hex : hexVector) {
+        sum = (sum << 4) | hex;
+        if(++count == rest) {
+            ss << bitsToBase32(sum);
+            rest += bitDivisionUnit;
+            sum = 0;
+        }
+    }
+
+    return ss.str();
+}
+
+int Euphony::Base32::convertChar2Int(char source) const {
+    switch(source) {
+        case '0': case '1': case '2':
+        case '3': case '4': case '5':
+        case '6': case '7': case '8':
+        case '9':
+            return source - '0';
+        case 'a': case 'b': case 'c':
+        case 'd': case 'e': case 'f':
+        case 'g': case 'h': case 'i':
+        case 'j': case 'k': case 'l':
+        case 'm': case 'n': case 'o':
+        case 'p': case 'q': case 'r':
+        case 's': case 't': case 'u':
+        case 'v':
+            return source - 'a' + 10;
+        default:
+            throw Base32Exception();
+    }
+}
+
+char Base32::convertInt2Char(int source) const {
+    const char base32Array[32] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
+                               'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j',
+                               'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't',
+                               'u', 'v'};
+
+    return base32Array[source];
+}
+
+const Euphony::HexVector &Euphony::Base32::getHexVector() const {
+    return hexVector;
+}
+
+std::string Base32::bitsToBase32(int value){
+    std::string result;
+    int count = 0;
+    int base32;
+
+    for (int i = 0; i < 20; i += 5) {
+        if ((value >> i) > 0) count = i / 5;
+    }
+
+    for (int i = count * 5; i >= 0; i -= 5) {
+        base32 = ((value >> i) & 31);
+        result.push_back(convertInt2Char(base32));
+    }
+
+    return result;
```

Comment:
```cpp
    std::string result = "";
    
    for (int i = 0; i < 20; i += 5) {
        int base32 = ((value >> i) & 31);
        result.append(convertInt2Char(base32));
    }

    return result;
```
It is better to reduce the number of iterations if possible. We can reduce like above pseudo code :)
Note that std::string has push_back like other STL functions, but there is also a convenient function called append.

Reply:
> It is better to reduce the number of iterations if possible.

thanks for your review!
I think so too. but it doesn't work the code above😢 It make calculation different.
I think we need to make general base32 & base64 which use '=' padding to reduce iteration. And if we use append, we need to cast ```convertInt2Char``` with <const char *> I think the way now is more simple!

Reply:
```suggestion
    string result;

    while(value != 0) {
        result = convertInt2Char(value & 31) + result;
        value >>= 5;
    }

    return result;
```

Reply:
@YoungSeokHong 
Thanks for your reply :)
I think above refactoring code will be run well :)

Reply:
Wow!! It is a really genius solution😱😄
I couldn't come up with how to add string to head. But it was really simple way! I need to stick with basics🤣🤣
I'm really appreciate about your solution!